src/mlpack/tests/data_facilities.hpp

```diff
@@ -0,0 +1,28 @@
+/**
+ * @file data_facilities.hpp
+ * @author Ryan Curtin, Kirill Mishchenko
+ *
+ * This file declares some useful data facilities.
```

Comment:
I think I actually added this file in the random forest PR as `mock_categorical_data.hpp`, but we can readapt the code in either case depending on what's merged first.